File: java/src/yax/test/CalcTest.java
package yax.test;

import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.CommandLineParser;
import org.apache.commons.cli.Options;
import org.apache.commons.cli.PosixParser;
import org.w3c.dom.Node;
import yax.*;

import java.io.FileReader;
import java.io.IOException;
import java.nio.charset.Charset;

public abstract class CalcTest
{

    //////////////////////////////////////////////////

    static public void
    main(String[] argv)
    {
        int flags = Util.FLAG_NONE;

        try {
	    CalcParser calcpushparser = null;
	    CalcEventHandler calceventhandler = null;

	    TestOptions.getoptions(argv);

            // 1. push parser
            calcpushparser = new CalcParser();
            if(TestOptions.parsetrace)
                calcpushparser.setDebugLevel(1);

            // 2. event handler
            calceventhandler = new CalcEventHandler(TestOptions.document, calcpushparser);
	    if(TestOptions.saxtrace)
		calceventhandler.setTrace(true);

            boolean ok = calceventhandler.parse();
            if(!ok)
                System.err.println("Parse failed");

        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        }
        System.err.println("No error");
        System.exit(0);
    }

    static String
    getinput(String filename)
        throws IOException
    {
        StringBuilder buf = new StringBuilder();
        FileReader file = new FileReader(filename);
        int c;

        while((c = file.read()) > 0) {
            buf.append((char) c);
        }
        return buf.toString();
    }
} // class CalcTest

File: python/yax/test/calctest.py
#This software is released under the terms of the Apache License version 2.
#For details of the license, see http://www.apache.org/licenses/LICENSE-2.0.

import os
import sys
import getopt
import codecs
import xml.sax
import StringIO

sys.path.append(os.path.join(os.getcwd(),"yax"))

from saxevent import SaxEvent
from saxeventhandler import SaxEventHandler
import saxeventtype
import util
import testoptions
from CalcParser import CalcParser
from calceventhandler import CalcEventHandler

def main() :
  testoptions.getOptions(sys.argv);

  # push parser
  calcpushparser = CalcParser()
  if (testoptions.parsetrace) :
    calcpushparser.setDebugLevel(1);

  # Create handler
  handler = CalcEventHandler(testoptions.document,calcpushparser)
  handler.setFlags(testoptions.flags)
  if(testoptions.saxtrace):
    handler.setTrace(True)

  # Parse the document
  handler.parse()

  sys.stdout.write("No error\n");
  sys.exit(0);
# end main()

if __name__ == '__main__':
  main()




